lib/supavisor.ex

```diff
@@ -80,6 +80,54 @@ defmodule Supavisor do
     end
   end
 
+  @doc """
+  During netsplits, or due to certain internal conflicts, :syn may store inconsistent data across the cluster.
+  This function terminates all connection trees related to a specific tenant.
+  It accomplishes this by going through all dynamic supervisor shards and searching for all supervisors
+  """
+  @spec dirty_terminate(String.t(), pos_integer()) :: map()
+  def dirty_terminate(tenant, timeout \\ 15_000) do
+    :erlang.whereis(Supavisor.DynamicSupervisor)
+    |> :supervisor.which_children()
```

Comment:
I am worried this request and traversal can be expensive when running thousands of clients. Maybe you should setup another registry (this one with duplicate keys) that stores all connections for a given tenant?

Reply:
Btw, you want to move these registries to the supervision tree:

https://github.com/supabase/supavisor/blob/d90556bbf8bc8971e9f940370edf52650badcdbb/lib/supavisor/application.ex#L55C1-L64C1

:)

It should be `{Registry, options}`. :)

Reply:
And to be clear, you won't use the `{:via, ...}` API to register those, but rather call Registry.register(...) directly to store all children for a tenant.

Reply:
My intention was to get supervisors from OTP internals to ensure that nothing was lost. But yes, it can be a heavy operation when there are a lot of connection trees. 

Take a look, please, at the updated commit when you have free time to ensure that I understand you correctly https://github.com/supabase/supavisor/pull/145/commits/1afe81a9720f9a1d8b4fde6b1345cd39aef92fa1

Reply:
That commit is perfect. 💯  :)

Reply:
Thanks for such a quick review!